controllers/forgotPassword_controller.js

```diff
@@ -6,17 +6,21 @@ const bcrypt = require("bcrypt");
 async function handleForgot(req, res) {
     try {
         req.body.email =  req.body.email.toLowerCase();
-        const user = dbHandler.getUserByEmail(req.body.email)
+        const user = await dbHandler.getUserByEmail(req.body.email);
+        if(!user){
+            throw new Error("user does'nt exist")
+        }
         const newPass = forgotPassService.generatePassword();
+
         await forgotPassService.sendPassword(newPass, user);
```

Comment:
modularity-code and reusability: 
consider moving the hashing part outside into a handler/ util